packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -193,17 +206,34 @@ async function updateRepoTeams(repo: Repo, context: Context) {
   console.log(`Update team access for ${repo.repo}`);
   const [owner, name] = repo.repo.split('/');
   const teamsToAdd = getRepoTeams(repo.language);
-  await Promise.all(
-    teamsToAdd.map(membership => {
-      return context.github.teams.addOrUpdateRepoInOrg({
-        team_slug: membership.slug,
-        owner,
-        org: owner,
-        permission: membership.permission as 'push',
-        repo: name,
-      });
-    })
-  );
+  try {
+    await Promise.all(
+      teamsToAdd.map(membership => {
+        return context.github.teams.addOrUpdateRepoInOrg({
+          team_slug: membership.slug,
+          owner,
+          org: owner,
+          permission: membership.permission as 'push',
+          repo: name,
+        });
+      })
+    );
+  } catch (err) {
+    const knownErrors = [
+      401, // bot does not have permission to access this repository.
```

Comment:
This is interesting!  Do you know what situation could lead to that?  Is it when someone installs the app, then we changed the required permissions after the fact?

Reply:
This method itself is kind of finicky... these are all the requirements:

- The authenticated user must have admin access to the repository, and must be able to see the team
- The repository must be owned by the organization, or a direct fork of a repository owned by the organization. 

I suspect it's caused by (1) failing for new repositories, as the bot may not have admin access to that repo.

Reply:
@sofisl I think you found that the bot needed to explicitly be added as an admin to some repos or something? is this an extra permission setting that's configurable?